de.metas.marketing/base/src/main/java/de/metas/marketing/base/model/CampaignRepository.java

```diff
@@ -150,4 +157,66 @@ private I_MKTG_Campaign createOrUpdateRecordDontSave(@NonNull final Campaign cam
 		campaignRecord.setMKTG_Platform_ID(campaign.getPlatformId().getRepoId());
 		return campaignRecord;
 	}
+
+	public int getDefaultNewsletterCampaignId(final int orgId)
+	{
+		return Services.get(IQueryBL.class).createQueryBuilder(I_MKTG_Campaign.class)
+				.addOnlyActiveRecordsFilter()
+				.addInArrayFilter(I_MKTG_Campaign.COLUMNNAME_AD_Org_ID, orgId, 0)
+				.orderByDescending(I_MKTG_Campaign.COLUMNNAME_AD_Org_ID)
+				.create()
+				.firstId();
+	}
+
+	public void removeContactPersonFromCampaign(
+			@NonNull final ContactPerson contactPerson,
+			@NonNull final Campaign campaign)
+	{
+		Services.get(IQueryBL.class).createQueryBuilder(I_MKTG_Campaign_ContactPerson.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_MKTG_Campaign_ContactPerson.COLUMN_MKTG_Campaign_ID, campaign.getCampaignId().getRepoId())
+				.addEqualsFilter(I_MKTG_Campaign_ContactPerson.COLUMN_MKTG_ContactPerson_ID, contactPerson.getContactPersonId().getRepoId())
+				.create()
+				.delete();
+
+	}
+
+	public void createConsent(
```

Comment:
this is about contactact persons and their consent, not about campaigns..please move this to ContactPersonRepository

Reply:
Done